tests/test_art_grpo.py

```
@@ -0,0 +1,180 @@
+            gpu_memory_utilization=0.3,
+        )
+
+        assert result["status"] == "success"
+
+        # The seed checkpoint should exist at the ART path
+        art_path = os.path.join(ckpt_dir, ".art")
+        assert os.path.isdir(art_path), f"ART path not found: {art_path}"
+
+        # Verify adapter_config.json exists somewhere in the checkpoints tree
+        found_adapter_config = False
+        for root, _dirs, files in os.walk(art_path):
```

Ruff (0.15.12)

[warning] Loop control variable `root` not used within loop body

Rename unused `root` to `_root`

(B007)